rest-api/dao/participant_dao.py

```diff
@@ -225,3 +246,24 @@ def _get_hpo_name_from_participant(participant):
 def raise_if_withdrawn(obj):
   if obj.withdrawalStatus == WithdrawalStatus.NO_USE:
     raise Forbidden('Participant %d has withdrawn' % obj.participantId)
+
+
+def make_primary_provider_link(hpo_name=None, hpo_id=None, hpo=None):
```

Comment:
The alternative to this is having make_primary_provider_link() wrapped by make_primary_provider_link_with_hpo_id() and make_primary_provider_link_with_hpo() helper methods that take one of those things...

I prefer that to having this looping conditional logic.

Reply:
Switched.